### PR TITLE
ASN1 Template: stricter and simpler DER/BER parsing/construction

### PR DESCRIPTION
**Pull request by SparkiDev**
Reduce debug output noise

### CODE REVIEW

**Review by dgarske — APPROVED** [review]
I've pushed updates for any issues found. I am happy with the state of it. Most changes are wrapped in the new WOLFSSL_ASN_TEMPLATE macro, which I did all of my testing with and various edge cases. Measured code size difference and almost none (slightly larger in some cases). Over to @ejohnstown

File: src/ssl.c [hunk 1]
```diff
@@ -40252,6 +40605,7 @@ int wolfSSL_i2d_X509_NAME(WOLFSSL_X509_NAME* name, unsigned char** out)
             goto cleanup;
         }
 
+        /* TODO: test case */
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply src/ssl.c:40608]: There is a test case now.
Removed comment.

File: src/ssl.c [hunk 2]
```diff
@@ -39881,7 +39899,13 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
     defined(WOLFSSL_CERT_GEN) && \
     (defined(WOLFSSL_CERT_REQ) || defined(WOLFSSL_CERT_EXT) || \
      defined(OPENSSL_EXTRA))
-/* Converts from NID_* value to wolfSSL value if needed */
+#ifndef WOLFSSL_ASN_TEMPLATE_SSLC
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply src/ssl.c:39902]: The ssl.c code got reworked.
I've fixed the code to not require ASN template code in ssl.c anymore.

**Comment by SparkiDev** [comment]
Rebased and merged all changes.

File: configure.ac [hunk 3]
```diff
@@ -174,6 +175,11 @@ AS_IF([test "$ax_enable_debug" = "yes"],
       [AM_CCASFLAGS="$DEBUG_CFLAGS $AM_CCASFLAGS"],
       [AM_CCASFLAGS="$AM_CCASFLAGS -DNDEBUG"])
 
+# COVERAGE
+AX_COVERAGE
```
Comment (ejohnstown) [inline configure.ac:179]: `AX_COVERAGE` isn't defined.

**Review by ejohnstown — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 3 above)
Comment (SparkiDev) [reply configure.ac]: Fixed - removed the lines.

**Review by ejohnstown — APPROVED** [review]

**Pull request merged**